accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/ui/AcToolUiService.java

```diff
@@ -0,0 +1,273 @@
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.text.SimpleDateFormat;
+import java.util.List;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang3.StringUtils;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import biz.netcentric.cq.tools.actool.api.InstallationLog;
+import biz.netcentric.cq.tools.actool.dumpservice.ConfigDumpService;
+import biz.netcentric.cq.tools.actool.history.AcHistoryService;
+import biz.netcentric.cq.tools.actool.history.AcToolExecution;
+import biz.netcentric.cq.tools.actool.history.PersistableInstallationLogger;
+import biz.netcentric.cq.tools.actool.impl.AcInstallationServiceInternal;
+
+@Component(service = { AcToolUiService.class })
+public class AcToolUiService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AcToolUiService.class);
+
+    public static final String PARAM_CONFIGURATION_ROOT_PATH = "configurationRootPath";
+    public static final String PARAM_APPLY_ONLY_IF_CHANGED = "applyOnlyIfChanged";
+    public static final String PARAM_BASE_PATHS = "basePaths";
+    public static final String PARAM_SHOW_LOG_NO = "showLogNo";
+    public static final String PARAM_SHOW_LOG_VERBOSE = "showLogVerbose";
+
+    public static final String PAGE_NAME = "actool";
+
+    static final String PATH_SEGMENT_DUMP = "dump.yaml";
+
+    @Reference(policyOption = ReferencePolicyOption.GREEDY)
+    private ConfigDumpService dumpService;
+
+    @Reference(policyOption = ReferencePolicyOption.GREEDY)
+    AcInstallationServiceInternal acInstallationService;
+
+    @Reference(policyOption = ReferencePolicyOption.GREEDY)
+    private WebConsoleConfigTracker webConsoleConfig;
+
+    @Reference(policyOption = ReferencePolicyOption.GREEDY)
+    private AcHistoryService acHistoryService;
+
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp, String postPath, boolean isTouchUi)
+            throws ServletException, IOException {
+
+        if (req.getRequestURI().endsWith(PATH_SEGMENT_DUMP)) {
+            streamDumpToResponse(resp);
+        } else {
+            renderUi(req, resp, postPath, isTouchUi);
+        }
+    }
+
+    protected void doPost(final HttpServletRequest req, final HttpServletResponse resp) throws IOException, ServletException {
+
+        RequestParameters reqParams = RequestParameters.fromRequest(req, acInstallationService);
+        LOG.info("Received POST request to apply AC Tool config with configurationRootPath={} basePaths={}",
+                reqParams.configurationRootPath, reqParams.basePaths);
+
+        InstallationLog log = acInstallationService.apply(reqParams.configurationRootPath, reqParams.getBasePathsArr(),
```

Comment:
Can't we explicitly pass a resourceResolver/usersession for apply?

Reply:
see https://github.com/Netcentric/accesscontroltool/issues/402#issuecomment-570176224